lib/compress.js

```diff
@@ -261,6 +262,24 @@ merge(Compressor.prototype, {
                         d.fixed = false;
                     }
                 }
+                if (node instanceof AST_Defun) {
+                    var d = node.name.definition();
+                    if (!toplevel && d.global || is_safe(d)) {
+                        d.fixed = false;
+                    } else {
+                        // Make a shallow copy of definition body but as
+                        // expression instead - ensure to clone whatever
+                        // child nodes you extract from this
```

Comment:
But in terms of implementation of new features, the current solution is not the best in that it could potentially trip people over.

I'm open to suggestions to make this more programmer-friendly whilst maintaining a good level of performance.